test/FunctionalTests/Infrastructure/GrpcTestFixture.cs

```diff
@@ -47,6 +47,8 @@ public GrpcTestFixture()
                 services.AddSingleton<DynamicGrpcServiceRegistry>();
 
                 services.AddSingleton<ILoggerFactory>(LoggerFactory);
+
+                services.AddSingleton<IPrimaryMessageHandlerProvider, TestPrimaryMessageHandlerProvider>(s => new TestPrimaryMessageHandlerProvider(_server));
```

Comment:
Injecting the TestServer into the TestServer

![image](https://user-images.githubusercontent.com/303201/53913849-28ddc880-40c1-11e9-83e7-0988d9f5122c.png)